pkg/webhook/podunavailablebudget/validating/pub_create_update_handler.go

```diff
@@ -0,0 +1,147 @@
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// PodUnavailableBudgetCreateUpdateHandler handles PodUnavailableBudget
+type PodUnavailableBudgetCreateUpdateHandler struct {
+	// To use the client, you need to do the following:
+	// - uncomment it
+	// - import sigs.k8s.io/controller-runtime/pkg/client
+	// - uncomment the InjectClient method at the bottom of this file.
+	Client client.Client
+
+	// Decoder decodes objects
+	Decoder *admission.Decoder
+}
+
+var _ admission.Handler = &PodUnavailableBudgetCreateUpdateHandler{}
+
+// Handle handles admission requests.
+func (h *PodUnavailableBudgetCreateUpdateHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	obj := &policyv1alpha1.PodUnavailableBudget{}
+
+	err := h.Decoder.Decode(req, obj)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	allErrs := h.validatingSidecarSetFn(obj)
+	if len(allErrs) != 0 {
+		return admission.Errored(http.StatusInternalServerError, allErrs.ToAggregate())
+	}
+	return admission.ValidationResponse(true, "")
+}
+
+func (h *PodUnavailableBudgetCreateUpdateHandler) validatingSidecarSetFn(obj *policyv1alpha1.PodUnavailableBudget) field.ErrorList {
+	//validate Pub.Spec
+	allErrs := validatePodUnavailableBudgetSpec(obj, field.NewPath("spec"))
+
+	//validate whether pub is in conflict with others
+	pubList := &policyv1alpha1.PodUnavailableBudgetList{}
+	if err := h.Client.List(context.TODO(), pubList, &client.ListOptions{Namespace: obj.Namespace}); err != nil {
+		allErrs = append(allErrs, field.InternalError(field.NewPath(""), fmt.Errorf("query other podUnavailableBudget failed, err: %v", err)))
+	} else {
+		allErrs = append(allErrs, validatePubConflict(obj, pubList.Items, field.NewPath("spec"))...)
+	}
+	return allErrs
+}
+
+func validatePodUnavailableBudgetSpec(obj *policyv1alpha1.PodUnavailableBudget, fldPath *field.Path) field.ErrorList {
+	spec := &obj.Spec
+	allErrs := field.ErrorList{}
+
+	if spec.Selector == nil && spec.TargetReference == nil {
+		allErrs = append(allErrs, field.Required(fldPath.Child("selector, targetRef"), "no selector or targetRef defined in PodUnavailableBudget"))
+	} else if spec.TargetReference != nil {
+		if spec.TargetReference.APIVersion == "" || spec.TargetReference.Name == "" || spec.TargetReference.Kind == "" {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("TargetReference"), spec.TargetReference, "empty TargetReference is not valid for PodUnavailableBudget."))
+		}
+	} else {
+		allErrs = append(allErrs, metavalidation.ValidateLabelSelector(spec.Selector, fldPath.Child("selector"))...)
+		if len(spec.Selector.MatchLabels)+len(spec.Selector.MatchExpressions) == 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("selector"), spec.Selector, "empty selector is not valid for PodUnavailableBudget."))
+		}
+	}
+
+	if spec.MaxUnavailable == nil && spec.MinAvailable == nil {
```

Comment:
should also validate `spec.MaxUnavailable != nil && spec.MinAvailable != nil` for they are mutually exclusive